pkg/obfuscator/ip.go

```diff
@@ -36,26 +44,51 @@ func (o *ipObfuscator) replace(s string) string {
 	ipv4matches := o.ipv4pattern.FindAllString(output, -1)
 	for _, m := range ipv4matches {
 		if ip := net.ParseIP(m); ip != nil {
-			output = strings.ReplaceAll(output, m, obfuscatedStaticIPv4)
-			o.ReportReplacement(m, obfuscatedStaticIPv4)
+			var replacement string
+			switch o.replacementType {
+			case schema.ObfuscateReplacementTypeStatic:
+				replacement = obfuscatedStaticIPv4
+			case schema.ObfuscateReplacementTypeConsistent:
+				if replacement = o.HasReplacement(m); replacement == "" {
+					o.ipv4count++
+					replacement = fmt.Sprintf(consistentIPv4Template, o.ipv4count)
+				}
+			default:
+				replacement = m
+			}
+			output = strings.ReplaceAll(output, m, replacement)
+			o.ReportReplacement(m, replacement)
 		}
 	}
 
 	ipv6matches := o.ipv6pattern.FindAllString(output, -1)
 	for _, m := range ipv6matches {
 		if ip := net.ParseIP(m); ip != nil {
-			output = strings.ReplaceAll(output, m, obfuscatedStaticIPv6)
-			o.ReportReplacement(m, obfuscatedStaticIPv6)
+			var replacement string
+			switch o.replacementType {
+			case schema.ObfuscateReplacementTypeStatic:
+				replacement = obfuscatedStaticIPv6
+			case schema.ObfuscateReplacementTypeConsistent:
+				if replacement = o.HasReplacement(m); replacement == "" {
+					o.ipv6count++
+					replacement = fmt.Sprintf(consistentIPv6Template, o.ipv6count)
+				}
+			default:
+				replacement = m
+			}
+			output = strings.ReplaceAll(output, m, replacement)
+			o.ReportReplacement(m, replacement)
```

Comment:
can you refactor those two V4+V6 code blocks into one method? I think you only differ in the use of the regex, the template and the counter

Reply:
+ same question for the default switch, probably should be an error